packages/auth/src/grant/service.ts

```diff
@@ -60,23 +68,44 @@ export async function createGrantService({
   const deps: ServiceDependencies = {
     logger: log,
     accessService,
+    config,
     knex
   }
   return {
     initiateGrant: (grantRequest: GrantRequest, trx?: Transaction) =>
       initiateGrant(deps, grantRequest, trx),
     getByInteraction: (interactId: string) => getByInteraction(interactId),
-    issueGrant: (grantId: string) => issueGrant(deps, grantId)
+    issueGrant: (grantId: string) => issueGrant(deps, grantId),
+    getByContinue: (
+      continueId: string,
+      continueToken: string,
+      interactRef: string
+    ) => getByContinue(continueId, continueToken, interactRef)
   }
 }
 
 async function issueGrant(
   deps: ServiceDependencies,
-  grantId: string
+  grantId: string,
+  trx?: Transaction
 ): Promise<Grant> {
-  return Grant.query(deps.knex).patchAndFetchById(grantId, {
-    state: GrantState.Granted
-  })
+  const invTrx = trx || (await Grant.startTransaction())
```

Comment:
nit pick: let's not call it `invTrx`. I think that still stands for "invoice transaction".

Also, what is the concurrent db transaction that we would pass here? I think what you had before was sufficient.